spark/src/test/scala/org/apache/spark/sql/delta/columnmapping/RemoveColumnMappingRowTrackingSuite.scala

```diff
@@ -21,65 +21,132 @@ import org.apache.spark.sql.delta.DeltaLog
+        // Update a row from each file to force materialize the Row IDs and metadata.
+        val predicate = s"$logicalColumnName % $rowsPerFile == 0"
+        withSQLConf(UPDATE_USE_PERSISTENT_DELETION_VECTORS.key -> "false") {
+          sql(s"UPDATE $testTableName SET $secondColumn = -1 WHERE $predicate ")
+        }
+
+        checkRowTrackingMarkedAsPreservedForCommit(deltaLog) {
+          testRemovingColumnMapping()
+        }
+      }
+    }
+  }
+
+  test("Row tracking marked as not preserved when row tracking disabled") {
+    withRowTrackingEnabled(enabled = false) {
+      withTable(testTableName) {
+        createTestTable()
+
+        val deltaLog = DeltaLog.forTable(spark, TableIdentifier(testTableName))
+
+        checkRowIdHighWaterMarkBeforeAndAfterOperation(deltaLog, diffRows = totalRows) {
+          checkRowIdsStayTheSameBeforeAndAfterOperation() {
+            assert(!rowTrackingMarkedAsPreservedForCommit(deltaLog) {
+              testRemovingColumnMapping()
+            })
+          }
+        }
+      }
+    }
+  }
+
+  test("Row tracking marked as not preserved when row tracking is supported " +
+    "but disabled") {
+    withRowTrackingEnabled(enabled = false) {
+      withTable(testTableName) {
+        createTestTable(tblProperties = Seq(s"'$rowTrackingFeatureName' = 'supported'",
+          s"'delta.minWriterVersion' = $TABLE_FEATURES_MIN_WRITER_VERSION"))
+
+        val deltaLog = DeltaLog.forTable(spark, TableIdentifier(testTableName))
+
+        checkRowIdHighWaterMarkBeforeAndAfterOperation(deltaLog, diffRows = totalRows) {
+          checkRowIdsStayTheSameBeforeAndAfterOperation() {
+            assert(!rowTrackingMarkedAsPreservedForCommit(deltaLog) {
+              testRemovingColumnMapping()
+            })
+          }
+        }
+      }
+    }
+  }
+  private def verifyRowIdsStayTheSame(originalRowIds: Array[Row]) = {
+    val newRowIds = spark.read.table(testTableName)
+      .select(logicalColumnName, RowId.QUALIFIED_COLUMN_NAME)
+    checkAnswer(newRowIds, originalRowIds)
+  }
+
+  private def createTestTable(tblProperties: Seq[String] = Seq.empty): Unit = {
+    val tblPropertiesStr = if (tblProperties.nonEmpty) {
+      s"TBLPROPERTIES ${tblProperties.mkString("(", ",", ")")}"
+    } else {
+      ""
+    }
     sql(
-      s"""CREATE TABLE $testTableName
+      s"""
+         |CREATE TABLE $testTableName
          |USING delta
-         |TBLPROPERTIES (
-         |'${DeltaConfigs.COLUMN_MAPPING_MODE.key}' = 'name',
-         |'${DeltaConfigs.ROW_TRACKING_ENABLED.key}' = 'true'
-         |)
+         |$tblPropertiesStr
          |AS SELECT id as $logicalColumnName, id + 1 as $secondColumn
          |  FROM RANGE(0, $totalRows, 1, $numFiles)
          |""".stripMargin)
+  }
 
-    val deltaLog = DeltaLog.forTable(spark, TableIdentifier(tableName = testTableName))
-
-    val snapshot = deltaLog.update()
+  private def checkRowIdsStayTheSameBeforeAndAfterOperation()(operation: => Unit): Unit = {
```

Comment:
Remove the empty parameter list:
```suggestion
  private def checkRowIdsStayTheSameBeforeAndAfterOperation(operation: => Unit): Unit = {
```

Use as:
```
checkRowIdsStayTheSameBeforeAndAfterOperation {
...
}
```